src/SmartComponents/AddSystemModal/AddSystemModal.js

```diff
@@ -43,12 +44,24 @@ export class AddSystemModal extends Component {
         });
     }
 
+    createContent = (id, content, body, name) => {
+        return {
+            id,
+            icon: <DriftTooltip
+                content={ content }
+                body={ body }
+            />,
+            name
+        };
+    }
+
     /*eslint-disable camelcase*/
     componentDidUpdate(prevProps) {
         const { baselines, handleBaselineSelection, handleHSPSelection, handleSystemSelection, historicalProfiles,
             selectedBaselineContent, selectedHSPContent, selectedSystemContent, systems } = this.props;
         let newSelectedSystems = [];
         let newSelectedBaselines = [];
+        let createContentFunc = this.createContent;
```

Comment:
@vkrizan I'll change that to a `const`. If I simply call `this.createContent()`, I get an error that it doesn't know what `this` is. This is essentially `undefined`.

Reply:
Oh, I see. Those inline function need to bind `this` in order to use it from the parent context/closure. One of the benefits of functional components is that the binding there isn't needed (as it does not need to reference the class).

It could look like this
```
newSelectedSystems = systems.map(function(system) {
    return this.createContentFunc(system.id, 'System', <ServerIcon />, system.display_name);
}.bind(this));
```